### DISCUSSION TITLE
SE_HW3_14022_Q1

### DISCUSSION THREAD

**Discussion by AmooHashem** [ssc-public/Software-Engineering#7] [Q&A]
سوالات خود پیرامون سوال اول از تمرین سوم درس در ترم ۱۴۰۲۲ را اینجا بپرسید!

**Comment by mehrafarin-k** [comment]
در قسمت اول سوال، ریسک های مدیریت هم جزو ریسک های طراحی نرم افزار حساب کنیم؟
اگر خیر ممکن است مورد دیگر جز Identifying risks.Prototypical risks. Prioritizing risks. رو بگید؟

**Reply by Saregh2000** [reply]
اگر مواردی باشن که تاثیری روی فرایند طراحی داشته باشن بله.

**Comment by mehrafarin-k** [comment]
قسمت ۵ سوال هم جوابش تو کتاب هست یا باید از منابع دیگه پیدا کنیم؟

**Reply by Saregh2000** [reply]
جوابتون ولید باشه رفرنسش کتاب باشه یا با سرچ منابع دیگه پیدا کرده باشید فرقی نداره

**Comment by kiarashk8128** [comment]
منظور از بخش دوم سوال یک که گفته دو مورد تکنیک را شرح دهید منظور همون هایی است که داخل جدول اومدن مثل Domain modeling, Security Analysis یا چیز های داخل متن کتاب مثل Techniques mitigate risks , ...

**Reply by Saregh2000** [reply]
سوال از روی اون جدول رفرنس طرح نشده ولی اگر جدول متدهاش ولیدن برای هدف صورت سوال میتونید استفاده کنید.

**Reply by Beny-Maleki** [reply]
آخه توی صورت سوال اشاره شده که «دو مورد از **_این_** تکنیک‌های مدیریت ریسک را تشریح کنید». چیزی که من برداشت می‌کنم اینه که نیازمندی سوال اینه که تکنیک‌های مدیریت ریسک توی **_مهندسی نرم‌افزار_** رو اشاره کنیم. توی کتاب هم دقیقا همین متد‌هایی که اشاره کردند رو به عنوان مثال‌هایی از تکنیک‌های کاهش ریسک در پروژه‌های مهندسی نرم‌افزار اشاره کرده بهشون. اگر منظور سوال چیز دیگه‌ای هست میشه دقیق‌تر بگید؟

**Reply by Beny-Maleki** [reply]
مثلا یک رویکرد دیگه‌ی این سوال که جنرال‌تر هست و محدود به پروژه‌های مهندسی نرم‌افزار نمیشه بررسی ROAM هست.
آیا اون مدنظره یا خیر؟

**Reply by Beny-Maleki** [reply]
اصلا چه توی این بخش از سوال و چه بخش اول سوال وقتی تکنیک‌ها قرار هست آزادانه توسط خودمون انتخاب بشن و نوشته بشن، در تصحیح به چه صورتی قرار هست که ریز به ریز صحیح بشوند؟ ممنون میشم این رو هم بیشتر توضیح بدید.

**Reply by Saregh2000** [reply]
ببینید این بخش های از سوال ۱ که مورد خواسته شده قطعا قرار نیست اینطوری صحیح بشه که یک جواب خاص داشته باشه. اگر توضیح و شرایط و ... دیگه ای برای هر مورد خواسته شده تلاش کنید تا جای ممکن ابعاد مختلف اون خواسته سوال رو بنویسید. برای مثال فقط به مطرح کردن تایتل یک ریسک نرم افزاری اکتفا نکنید. احتمال رخدادش و اگر امکانش رو دارید سناریوی رخداد و اثراتی که میذاره و ... را هم بنویسید یا برای متد های مدیریت تلاش کنید جنبه های مختلف متد رو انالیز کنید و به اختصار اشاره کنید.

**Comment by Beny-Maleki** [comment]
توی بخش اول سوال یک گفته شده باید ۴ مورد ریسک‌های ممکن رو شرح بدیم بعد در انتهای سوال گفته شده به دنبال identifying risks و prototypical risks و ... بریم.
این مورد اول اصلا نوعی از ریسک نیست. به طور کلی این قسمتی از کتاب هم که اشاره کردید بهش دسته‌بندی مشخصی از ریسک انجام نداده و صرفا یک سری مثال روزمره از ریسک رو در خودش داره. با توجه به ریز بودن تصحیح لطفا منظور از این قسمت رو دقیق‌تر بیان کنید.

**Reply by Saregh2000** [reply]
این موارد گفته شده یک سری مثال از خانواده ریسک های ممکن هستن شما میتونید برید سمت دسته های دیگه مثل performance و ui/ux و انواع دیگه ریسک ها الزامی نیست مطالب حتما از تو کتاب جواب داده بشن. برای این مورد که ریز صحیح میشه هم سعی کنید کامل توضیحات مربوط به شرایط منجر به ریسک حتی اگر پیدا کردید سلوشن و این اطلاعات رو اشاره بکنید حتما.

**Reply by Beny-Maleki** [reply]
تشکر
من منظورم این هست که با توجه به توضیحاتی که برای بخش‌های describing risks و identifying risks دارم می‌خونم به نظر میاد که این موارد در اصل روش‌هایی برای صیقل‌کاری ریسک‌های سیستم هستند و اینکه چطور بتونیم در ادامه ریسک‌ها رو مدیریت کنیم. در اصل نه تکنیک به حساب میان و نه نوعی از ریسک. جنس pre-process دارند. و خب با توجه به چیزی که بخش الف خواسته از ما نمی‌دونم که کجای از جوابم باید از اینا استفاده کنم.

**Reply by Saregh2000** [reply]
خوب بذار یک شفاف سازی بکنم. اینجا منظور از identifying risks اون کانسپتی که تو کتاب rdd اومده نیست. مشخصا به خانواده ای از ریسک ها اشاره میکنه که ممکنه در فرایند استخراج نیازمندی ها یا خود مرحله انالیز ریسک های پروژه رخ دهند. امیدوارم شفاف شده باشه.
